models/users.js

```diff
@@ -1,15 +1,44 @@
-const mongoose = require('mongoose')
+const mongoose = require( 'mongoose' )
 const Schema = mongoose.Schema
+const bcrypt = require( 'bcrypt-nodejs' )
 
 const UserSchema = new Schema({
-  name: String,
-  username: { type: String, required: true, unique: true },
-  password: { type: String, required: true },
-  location: String,
-  created_at: Date,
-  updated_at: Date
+   email: { type: String, required: true, unique: true },
+   password: { type: String, required: true },
+   token: { type: String },
+   token_expires: { type: Date },
+   created_at: Date,
+   updated_at: Date
 })
 
-const User = mongoose.model('User', UserSchema)
+UserSchema.pre( 'save', function( next ){
+  const user = this
+  const SALT_FACTOR = 5
 
-module.exports = User
+  if (!user.isModified( 'password' )){
+    return next()
+  }
+
+  bcrypt.genSalt( SALT_FACTOR, ( err, salt ) => {
+    if (err)
+      return next(err)
+
+    bcrypt.hash( user.password, salt, null, ( err, hash ) => {
+      if (err)
```

Comment:
Please add spaces between opening and closing parentheses, and arguments, in a function invocation or control flow expression:
```js
fn(param1, param2) { /* ... */ }
if(expression) { /* ... */ }
```
should be
```js
fn( params1, param2 ) { /* ... */ }
if( expression ) { /* ... */ }
```
Please do not use single line control flow, and always include braces:
```js
if(expression) return thing;
```
should be
```js
if( expression ) {
  return thing;
}
```